### PR TITLE
Update Verilator version

### PR DESCRIPTION
**Pull request by vogelpi**
As of version 4.028, Verilator supports wildcard matching for `lint_off` rules. This reduces the effort required to keep lint waiver files aligned with the actual source code.

### CODE REVIEW

**Comment by imphil** [comment]
Please also add the machinery we have in Ibex to warn users if they use an outdated Verilator version. We currently don't have a reliable way to inform downstream users of such changes they need to make to their environment, so this is our only chance (or we get grumpy users).

File: util/build_docs.py [hunk 1]
```diff
@@ -196,6 +196,23 @@ def generate_apt_reqs():
...
+    The version number per tool will be saved in outdir-generated/version_$TOOL_NAME.txt
+    """
+
+    # Populate __TOOL_REQUIREMENTS__
+    requirements_file = str(SRCTREE_TOP.joinpath("tool_requirements.py"))
+    exec(open(requirements_file).read(), globals())
+
+    # And then write a version file for every tool.
+    for tool in __TOOL_REQUIREMENTS__:
+        version_path = config["outdir-generated"].joinpath('version_' + tool + '.txt')
```
Comment (imphil) [inline util/build_docs.py]:
```suggestion
        version_path = config["outdir-generated"] / 'version_' + tool + '.txt'
```

**Comment by vogelpi** [comment]
@imphil I managed to port your framework from Ibex to OpenTitan (the hardest part was actually to also use it with Hugo).

For now, the check is just included for linting via ` hw/lint/common.core`. I am not sure if this is sufficient. But before touching many .core files, I would like to get your input if this goes in the right direction.

**Comment by vogelpi** [comment]
Ah, I just found out that lint is also called before generating e.g. top_earlgrey. So this should actually be sufficient. WDYT?

File: util/build_docs.py (hunk 1 above)
Comment (vogelpi) [reply util/build_docs.py]: Thanks for your feedback @imphil . Your suggestion does not work as the first part is of type `PosixPath` and the second part is `str`. We are using `joinpath()` in several places in this file. I thus suggest to leave it.

**Comment by vogelpi** [comment]
I made the new shortcode more generic: the tool name is now passed as an arugment to the shortcode `{{< tool_version "verilator" >}}` . Meaning we can use the same shortcode also for other tools in the future.

File: util/build_docs.py (hunk 1 above)
Comment (imphil) [reply util/build_docs.py]: '/' also works with strings, but operator precedence requires parentheses around the string concatenation with `+`. So `config["outdir-generated"] / ('version_' + tool + '.txt')` or `config["outdir-generated"] / 'version_%s.txt' % tool` should do the trick.

No need to change that now, but to keep in mind for future changes.

**Pull request merged**